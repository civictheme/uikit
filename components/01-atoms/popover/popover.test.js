const template = 'components/01-atoms/popover/popover.twig';

describe('Popover Component', () => {
  test('renders with required attributes', async () => {
    const c = await dom(template, {
      trigger: {
        text: 'Sample Trigger',
      },
      content: 'Sample content',
    });

    expect(c.querySelectorAll('.ct-popover')).toHaveLength(1);
    expect(c.querySelector('.ct-popover__link').textContent.trim()).toEqual('Sample Trigger');
    expect(c.querySelector('.ct-popover__content').textContent.trim()).toEqual('Sample content');

    assertUniqueCssClasses(c);
  });

  test('renders with optional attributes', async () => {
    const c = await dom(template, {
      trigger: {
        text: 'Sample Trigger',
        url: 'https://example.com',
        is_new_window: true,
        is_external: true,
      },
      content: 'Sample content',
      group: 'sample-group',
      theme: 'dark',
      attributes: 'data-test="true"',
      modifier_class: 'custom-class',
    });

    expect(c.querySelectorAll('.ct-popover.custom-class.ct-theme-dark')).toHaveLength(1);
    expect(c.querySelector('.ct-popover__link').textContent.trim()).toContain('Sample Trigger');
    expect(c.querySelector('.ct-popover__link').getAttribute('href')).toEqual('https://example.com');
    expect(c.querySelector('.ct-popover__link').getAttribute('target')).toEqual('_blank');
    expect(c.querySelector('.ct-popover').getAttribute('data-test')).toEqual('true');
    expect(c.querySelector('.ct-popover__content').textContent.trim()).toEqual('Sample content');
    expect(c.querySelector('.ct-popover').getAttribute('data-collapsible-group')).toEqual('sample-group');

    assertUniqueCssClasses(c);
  });

  test('does not render when content is empty', async () => {
    const c = await dom(template, {
      trigger: {
        text: 'Sample Trigger',
      },
      content: '',
    });

    expect(c.querySelectorAll('.ct-popover')).toHaveLength(0);
  });
});
