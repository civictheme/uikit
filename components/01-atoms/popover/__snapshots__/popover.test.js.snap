// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Component does not render when content is empty 1`] = `
<div>
  



</div>
`;

exports[`Popover Component renders with optional attributes 1`] = `
<div>
  


  
  <div
    class="ct-popover ct-theme-dark custom-class"
    data-collapsible=""
    data-collapsible-collapsed=""
    data-collapsible-duration="250"
    data-collapsible-group="sample-group"
    data-test="true"
  >
    
    



  
        
  
  
    <a
      aria-label="Opens in a new tab"
      class="ct-link ct-theme-dark     ct-popover__link"
      data-collapsible-trigger=""
      href="https://example.com"
      tabindex="0"
      target="_blank"
    >
      
              Sample Trigger
            
      <span
        class="ct-visually-hidden"
      >
        (Opens in a new tab/window)
      </span>
      
    
  
    </a>
    

    
    <div
      class="ct-popover__content"
      data-collapsible-panel=""
    >
      
      
      <div
        class="ct-popover__content__inner"
      >
        

                          
                  Sample content
        
                                
      </div>
      
    
    </div>
    
  
  </div>
  

</div>
`;

exports[`Popover Component renders with required attributes 1`] = `
<div>
  


  
  <div
    class="ct-popover ct-theme-light "
    data-collapsible=""
    data-collapsible-collapsed=""
    data-collapsible-duration="250"
    data-collapsible-group=""
  >
    
    



  
  
  
  
    <a
      class="ct-link ct-theme-light     ct-popover__link"
      data-collapsible-trigger=""
      tabindex="0"
    >
      
              Sample Trigger
      
  
    </a>
    

    
    <div
      class="ct-popover__content"
      data-collapsible-panel=""
    >
      
      
      <div
        class="ct-popover__content__inner"
      >
        

                          
                  
        <span>
          Sample content
        </span>
        
        
                                
      </div>
      
    
    </div>
    
  
  </div>
  

</div>
`;
